src/osp/UserInputHandler.cpp

```diff
@@ -245,8 +245,15 @@ void UserInputHandler::config_register_control(std::string const& name,
         std::initializer_list<ButtonVarConfig> vars)
 {
     std::vector<ButtonVarConfig> varVector(vars);
-    m_controlConfigs[name] = ButtonConfig{std::move(varVector), holdable,
-                                          false, 0};
+    config_register_control(name, holdable, std::move(varVector));
+}
+
+void UserInputHandler::config_register_control(std::string const& name,
+    bool holdable,
+    std::vector<ButtonVarConfig> vars)
+{
+    m_controlConfigs[name] = ButtonConfig{ std::move(vars), holdable,
```

Comment:
use std::map::insert_or_assign for this.

Reply:
and you should also make two instances of the function.

one that takes the `name` parameter by const&, and the other by &&. That way we can get the speed gains of move semantics when the key is an rvalue reference.

Note that i say to do this way instead of taking the parameter by value, specifically because the key may not actually be moved from if the item already exists in the map, so we don't want to pay the cost of making a copy of the string if we don't need to.